data_structures/queue_using_array.cpp

```diff
@@ -1,77 +1,128 @@
-/*
-    Write a program to implement Linear Queue using array.
+/**
+ * @file
+ * @brief Implementation of Linear [Queue using array]
+ * (https://www.geeksforgeeks.org/array-implementation-of-queue-simple/).
+ * @details
+ * The Linear Queue is a data structure used for holding a sequence of
+ * values, which can be added to the end line (enqueue), removed from
+ * head of line (dequeue) and displayed.
+ * ### Algorithm
+ * Values can be added by increasing the `rear` variable by 1 (which points to 
+ * the end of the array), then assigning new value to `rear`'s element of the array.
+ * 
+ * Values can be removed by increasing the `front` variable by 1 (which points to 
+ * the first of the array), so it cannot reached any more.
+ * 
+ * @author [Pooja](https://github.com/pooja-git11)
+ * @author [Farbod Ahmadian](https://github.com/farbodahm)
+ */
+#include <iostream> /// for io operations
+#include <vector>   /// for std::vector
 
-    Functions to implement
-        Enqueue (Insertion)
-        Dequeue (Deletion)
+constexpr uint16_t max_size{10}; ///< Maximum size of the queue
```

Comment:
@farbodahm Great improvement over documentation and code fixes, Thanks for your contribution.
Here is a suggestion, basically vectors are used to store the queue elements so there is no need of queue size, it would be great if you do a little imrpovement over this.

Reply:
Thanks for your great review,

If we don't use a variable to store `max size` and just rely on `std::vector`, then this code will not be an implementation for **linear** queue because it can grow as much as user want.
So we can use `std::array` instead of `std::vector` to have `linear` queue OR we can still use `std::vector` and delete `max_size` variable but it will not be a linear queue any more.

Just let me know which one is preferred, and I will change that as you wished :)

Reply:
i think it's good to use arrays because the idea behind the implementation of these data structures is to give the foundational idea of how the standard data structures like `std::list` or `std::deque` are implemented on top of basic arrays.
And as here we are dealing with C++ it's good to use `std::array` instead of raw C style arrays.
Hope this makes sense.